Src/Fido2.Models/CredentialCreateOptions.cs

```diff
@@ -205,6 +212,7 @@ public class AuthenticatorSelection
         public static AuthenticatorSelection Default => new AuthenticatorSelection
         {
             AuthenticatorAttachment = null,
+            ResidentKey = ResidentKeyRequirement.Required,
```

Comment:
I'm good with the resident key requirement enum, but why do we want the default to be required?  This feels problematic.

Reply:
I chose it to be Required based on following sentence in the [standard](https://w3c.github.io/webauthn/#dictionary-authenticatorSelection):

> **If no value is given then the effective value is [required](https://w3c.github.io/webauthn/#dom-residentkeyrequirement-required) if [requireResidentKey](https://w3c.github.io/webauthn/#dom-authenticatorselectioncriteria-requireresidentkey) is true** or [discouraged](https://w3c.github.io/webauthn/#dom-residentkeyrequirement-discouraged) if it is false or absent.

But my interpretation was wrong. The default value should be `discouraged` because `requireResidentKey` is default `false` and an absent `residentKey` property would then be defaulted to `discouraged`.

If I think further about the connection these two properties have, then changing one should change the other. For example it makes no sense for a user of the library to set `requireResidentKey=true and residentKey=discouraged`
I see two possible approaches.

Non breaking approach:
- Set requireResidentKey to true sets residentKey to required
- Set requireResidentKey to false sets residentKey to discouraged
- Set residentKey to discouraged sets requireResidentKey to discouraged
- Set residentKey to preferred sets requireResidentKey to discouraged (requireResidentKey: [Relying Parties](https://w3c.github.io/webauthn/#relying-party) SHOULD set it to true if, and only if, [residentKey](https://w3c.github.io/webauthn/#dom-authenticatorselectioncriteria-residentkey) is set to [required](https://w3c.github.io/webauthn/#dom-residentkeyrequirement-required).)
- Set residentKey to required sets requireResidentKey to true

Breaking approach but better because of no redundant properties:
- Set property `RequireResidentKey` to private and apply the same mapping as specified above.
- Set residentKey to discouraged sets requireResidentKey to discouraged
- Set residentKey to preferred sets requireResidentKey to discouraged (requireResidentKey: [Relying Parties](https://w3c.github.io/webauthn/#relying-party) SHOULD set it to true if, and only if, [residentKey](https://w3c.github.io/webauthn/#dom-authenticatorselectioncriteria-residentkey) is set to [required](https://w3c.github.io/webauthn/#dom-residentkeyrequirement-required).)
- Set residentKey to required sets requireResidentKey to true

Which option would you prefer that i implement @aseigler?

Reply:
What I am not sure is, if we could remove requireResidentKey completely.
About requireResidentKey: 

> This member is retained for backwards compatibility with WebAuthn Level 1...

Does the specification describe it for backward compatibility or should the RP include it for backward compatibility...

Reply:
I am struggling to follow you on the the pr 227. Could it be that you linked the wrong one?
I checked the webauthntest and saw that they allow choosing "Require Resident Key" and "Resident Key (L2)". I think as well that it is elegant to include only selected options using the `!==` operator in js. But I don't think that it is useful to include both properties for the demo in general. Require Resident Key is retained for backwards compatibility and is specified definite through ResidentKey. I see no functional need to include both. It only opens the possibility to choose incompatible values. Could you give me a clearer hint what you would do different?

Reply:
Oof, yes, I meant #277. I agree backward compatibility doesn't really make any sense to keep for the demo for this property, just makes things more confusing.

Reply:
Oh ok. I see that you included requireResidentKey and residentKey in the demo in this branch. But you didn't integrate it into the c# code. The second property (residentKey) is set in the javascript code to `makeCredentialOptions.authenticatorSelection` after the makeCredentialOptions have been fetched from the server.

I see two options.
I integrate the code of this pull request into 277, but i will remove the `requireResidentKey` option, and i will keep the c# integration, so that in all the c# and javascript code only `residentKey` is used.
Or we merge this pullrequest into the master and then i will merge the master into your demo branch and do the same clean up. What would suit you better? What is the timeline for the demo branch? :)